cartridge/failover/raft.lua

```diff
@@ -66,9 +66,9 @@ local function on_election_trigger()
     local box_info = box.info()
     local election = box_info.election
 
-    local leader = box_info.replication[election.leader] or {}
+    local leader = box_info.replication[election.leader]
 
-    if vars.leader_uuid ~= leader.uuid then
+    if leader ~= nil and vars.leader_uuid ~= leader.uuid then
```

Comment:
If a check is added here for the presence of a elected leader exists, then the payload of raft term will be updated, but the payload leader will not be updated. This will cause an algorithm (in function get_appointments) that relies on the fact that a newer term contains up-to-date data will actually get outdated data in this case.

Reply:
https://github.com/tarantool/cartridge/issues/1900#issuecomment-1297085086

Reply:
@R-omk Hi! I updated the code to check terms and leaders. Does it fix your problem?

Reply:
If by "problem" you mean issue1900 then I'm not sure, because leader_uuid is only set to member in a trigger that might never have been executed after the cartridge was started.

In terms of raft info consistency, it looks better, but it's still better to keep the payload as one indivisible object, and deal with the data type confusion that I wrote about in another comment.